OpenRA.Mods.Common/Traits/World/EditorResourceLayer.cs

```diff
@@ -188,37 +187,34 @@ public int CalculateRegionValue(CellRegion sourceRegion)
 			foreach (var cell in sourceRegion.CellCoords)
 			{
 				var mcell = cell.ToMPos(Map);
-				if (Map.Resources.Contains(mcell) && Map.Resources[mcell].Type != 0)
-				{
-					resourceValueInRegion++;
-					var rcell = Map.Resources[mcell];
-					if (ResourceTypesByIndex.TryGetValue(rcell.Type, out var resourceType) && resourceValues.TryGetValue(resourceType, out var resourceValuePerUnit))
-						resourceValueInRegion += Tiles[mcell].Density * resourceValuePerUnit;
-				}
+				byte resource;
+				if (Map.Resources.Contains(mcell)
+					&& (resource = Map.Resources[mcell].Type) != 0
+					&& ResourceTypesByIndex.TryGetValue(resource, out var resourceType)
+					&& resourceValues.TryGetValue(resourceType, out var resourceValuePerUnit))
+					resourceValueInRegion += Tiles[mcell].Density * resourceValuePerUnit;
 			}
 
 			return resourceValueInRegion;
 		}
 
-		protected virtual int CalculateCellDensity(ResourceLayerContents contents, CPos c)
+		protected virtual int CalculateCellDensity(ResourceLayerContents contents, CPos cell)
 		{
 			var resources = Map.Resources;
-			if (contents.Type == null || !info.ResourceTypes.TryGetValue(contents.Type, out var resourceInfo) || resources[c].Type != resourceInfo.ResourceIndex)
+			if (contents.Type == null || !info.ResourceTypes.TryGetValue(contents.Type, out var resourceInfo) || resources[cell].Type != resourceInfo.ResourceIndex)
 				return 0;
 
 			if (!info.RecalculateResourceDensity)
 				return contents.Density.Clamp(1, resourceInfo.MaxDensity);
 
 			// Set density based on the number of neighboring resources
 			var adjacent = 0;
-			for (var u = -1; u < 2; u++)
+			var directions = CVec.Directions;
+			for (var i = 0; i < directions.Length; i++)
 			{
-				for (var v = -1; v < 2; v++)
-				{
-					var cell = c + new CVec(u, v);
-					if (resources.Contains(cell) && resources[cell].Type == resourceInfo.ResourceIndex)
-						adjacent++;
-				}
+				var c = cell + directions[i];
+				if (resources.Contains(c) && resources[c].Type == resourceInfo.ResourceIndex)
+					++adjacent;
 			}
 
 			return Math.Max(int2.Lerp(0, resourceInfo.MaxDensity, adjacent, 9), 1);
```

Comment:
Can we include the comment `// Adjacent includes the current cell, so is always >= 1` to match `ResourceLayer`?

Reply:
The point of this PR is that adjacent does not include the current cell. That it was included for the map editor, but actually was not included in `ResourceLayer`, that the whole premise of #13318 was wrong, including this comment

Reply:
Can we please remove the comment from `ResourceLayer` then?

Reply:
updated

Reply:
I've added a comment specifying that not including the current cell is an error